### ISSUE TITLE
Next steps after running training mode

### ISSUE THREAD

**Issue by raitiss5** [liaoherui/GDmicro#6]
Hi,
I apologize for the amateur question but I ran the GDmicro model on my dataset in training mode:
python GDmicro.py -i Input_files/AD_inputgdmicro_train.csv -b 32 -d AD -t 1 -o GDmicro_Res/ADtrain

Now, I would like to test it on a separate test dataset to classify disease status, how can I do that using the trained model?
Thanks.

**Comment by liaoherui** [comment]
Hi, thanks for using GDmicro!

To predict disease status for your test data, please include your test samples in the microbial species abundance matrix (see details below). An example input file containing both training and test datasets can be found at: Input_files/CRC_CHI_LOSO.csv.

However, currently, we have changed some codes/functions to make the tool be more useful in identifying microbial biomarkers rather than classification. Thus, the performance may be influenced (may be decreased) but the biomarker discovery function should work better. 


<img width="873" alt="Image" src="https://github.com/user-attachments/assets/50ee6228-f657-4055-9648-46c8cc6ce6de" />

**Comment by raitiss5** [comment]
Hi, thank you for your reply.

I wanted to clarify, after training on the samples I gave during the training mode step, is the trained model saved? Or is running step 1.1 as you've highlighted in the image the same as the training step in 1.2 with the additional step of testing as well?

Additionally, if the model is trained separately (step 1.2), is there a way to save and reuse it later, or does it need to be retrained each time?

I'll keep what you've said about the biomarker discovery function in mind!

**Comment by liaoherui** [comment]
"Or is running step 1.1 as you've highlighted in the image the same as the training step in 1.2 with the additional step of testing as well?" → Yes. You should run step 1.1 with both your training and additional test samples.

"Or does it need to be retrained each time?" → Given this is a GCN model, every time you add test samples to the graph, you get a new graph containing both training and new test samples, thus, you need to rerun the model.

Again, because this is a GCN model, unlike CNN or Transformer models, it takes a graph as input. You can imagine that even if you have a pretrained model, it cannot be directly applied to your new test samples, because you must create a new graph that includes these samples. 

Feel free to discuss if you have any thoughts or suggestions on this! (In fact, I am also interested and would be happy to develop a pretrained model approach for GCN/GDmicro if there's a good solution!)

**Comment by raitiss5** [comment]
I'm still quite new to deep learning and machine learning in general, so my input might not be anything new for you - have you considered experimenting with inductive GCNs like the GraphSAGE framework? From what I've gathered reading forum posts, GCNs can conduct inductive learning if sampling is available? What about training a GCN to learn node embeddings, then using a separate classifier like an MLP to classify new samples without needing the graph at test time?

Please do let me know why they could/wouldn't work as I have much to learn.

**Comment by liaoherui** [comment]
Thanks for these inspiring suggestions! 

I've tried GraphSAGE before but ultimately couldn't incorporate it into GDmicro for some reasons (however, I can not remeber the details now...). Your second solution sounds very interesting, and I haven't explored it yet. I'll definitely give it a try when I have some availability, though I'm currently occupied with several ongoing projects... 

Again, I appreciate these informative solutions!

**Comment by raitiss5** [comment]
Glad I could give some input! It would be awesome if you could get a workaround for this, I think it'd be a really valuable addition to GDmicro. Good luck!
